internal/operator-controller/rukpak/bundle/config.go

```diff
@@ -6,119 +6,489 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/santhosh-tekuri/jsonschema/v6"
 	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/apimachinery/pkg/util/validation"
 	"sigs.k8s.io/yaml"
 
 	"github.com/operator-framework/api/pkg/operators/v1alpha1"
 )
 
+const (
+	// configSchemaID is a name we use to identify the config schema when compiling it.
+	// Think of it like a file name - it just needs to be consistent.
+	configSchemaID = "config-schema.json"
+
+	// OwnNamespace mode: watchNamespace must equal install namespace
+	formatOwnNamespaceInstallMode = "installNamespace"
+	// SingleNamespace mode: watchNamespace must differ from install namespace
+	formatSingleNamespaceInstallMode = "notInstallNamespace"
+)
+
+// ConfigSchemaProvider is how bundles tell us their configuration rules.
+//
+//   - registry+v1 bundle -> generates schema from the operator's install modes
+//   - Helm chart -> reads schema from values.schema.json in the chart
+//   - registry+v2 bundle -> (future) will provide schema via its own mechanism
+type ConfigSchemaProvider interface {
+	// GetConfigSchema returns a JSON schema describing valid configuration.
+	//
+	// Example schema:
+	//   {
+	//     "type": "object",
+	//     "required": ["watchNamespace"],
+	//     "properties": {
+	//       "watchNamespace": {"type": "string"}
+	//     }
+	//   }
+	GetConfigSchema() (string, error)
```

Comment:
can we shorten the name to `Get()`? given that the interface is already called `ConfigSchemaProvider`?

Reply:
I think it is solved now. 
Feel free to resolve if you agree.